packages/fuels-core/src/abi_encoder.rs

```diff
@@ -66,19 +66,23 @@ impl ABIEncoder {
         self.encode_tokens(arg_array)
     }
 
-    fn encode_string(&mut self, arg_string: &StringToken) {
-        if !arg_string.0.is_ascii() {
-            panic!("String parameters can only have ascii values");
+    fn encode_string(&mut self, arg_string: &StringToken) -> Result<(), CodecError> {
+        if !arg_string.data.is_ascii() {
```

Comment:
Looking at this I'm wondering two things:
1. Can we extract these two checks into `impl StringToken`?
2. Better yet, can we make StringToken conditionally constructible?

I'm not sure of the different ways tokens are used but I have a feeling that once you have a Token::String on your hands you're not going to change its innards but rather pass it around. 

If this is so then we can effectively treat `StringToken` as an immutable data structure -- make its innards private, have a `new` method that will return `Result<StringToken, Error>` and have the assurance that if we managed to get our hands on a `StringToken` the string inside is definitely valid.

Thoughts?

Reply:
I have implemented option 1. Regarding option 2, we can not implement it without completely changing how the contract methods are generated and I do not recommend that.